app/main.py

```diff
@@ -1,2 +1,146 @@
+from __future__ import annotations
+from typing import Any, Hashable
+
+INITIAL_CAPACITY = 8
+LOAD_FACTOR = 2 / 3
+RESIZE_FACTOR = 2
+
+
+class Node:
+    def __init__(self, key: Hashable, value: Any) -> None:
+        self.key = key
+        self.value = value
+        self.hash_key = hash(key)
+
+
 class Dictionary:
-    pass
+
+    def __init__(self, **kwargs: Any) -> None:
+        self.length = 0
+        self.dict_keys: list = []
+        self.dict_values: list = []
+        self.capacity: int = INITIAL_CAPACITY
+        self.hash_table: list = [None] * self.capacity
+        for key, value in kwargs.items():
+            self.__setitem__(key, value)
+
+    def __str__(self) -> str:
+        result_str = "{"
+
+        for i in range(len(self.hash_table)):
+
+            if self.hash_table[i] is not None:
+                result_str += (f"'{self.hash_table[i].key}': "
+                               f"{self.hash_table[i].value}, ")
+
+        return f"{result_str}"[:-2] + "}"
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        node = Node(key, value)
+        index = node.hash_key % self.capacity
+
+        if (index == len(self.hash_table)
+                and self.hash_table[index] is not None
+                and self.hash_table[index].key != key):
+            index = 0
+
+        while (self.hash_table[index] is not None
+               and self.hash_table[index].key != key):
+            index = (index + 1) % self.capacity
+
+        if self.hash_table[index] is None:
+            self.length += 1
+
+        self.hash_table[index] = node
+
+        if self.length / self.capacity >= LOAD_FACTOR:
+            self._resize()
+
+    def __getitem__(self, key: Hashable) -> Any:
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+
+        while (self.hash_table[index] is not None
+               and self.hash_table[index].key != key):
+            index = (index + 1) % self.capacity
+
+        if self.hash_table[index] is None:
+            raise KeyError(key)
+
+        return self.hash_table[index].value
+
+    def __len__(self) -> int:
+        return self.length
+
+    def __delitem__(self, key: Hashable) -> None:
+        for i in range(self.length):
+            if (self.hash_table[i] is not None
+                    and self.hash_table[i].key == key):
+                self.hash_table[i] = None
```

Comment:
In this method, you could use the same loop condition as in the __getitem__ method to find the slot that contains the key, instead of iterating over the entire hash table, like this + it's good to raise exception(not sure if code works, its just fastprint):
def delitem(self, key):
        key_hash = hash(key)
        index = key_hash % self._capacity

        while (self._hash_table[index] is not None
               and self._hash_table[index].key != key):
            index = (index + 1) % self._capacity

        if self._hash_table[index] is None:
            raise KeyError(key)

        self._hash_table[index] = None
        self._length -= 1

Reply:
+